### PR TITLE
[proof] Make LeafCount type alias

### PR DESCRIPTION
**Pull request by zamsden**
## Motivation

Add a type alias for LeafCount as u64 so that 32-bit builds will
continue to work.  This is required for external HSM code that
needs to include types crate.  We lose some type checking but
gain readability and consistency on what is a Version versus
what is a LeafCount.

### Have you read the [Contributing Guidelines on pull requests]

Ja, ek het dit gelees.

## Test Plan

CI

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by zamsden** [comment]
Toolchain update left me without rustfmt :(  Fixed.

File: types/src/proof/position/position_test.rs [hunk 1]
```diff
@@ -259,27 +259,27 @@ fn slow_get_frozen_subtree_roots_impl(root: Position, max_leaf_index: u64) -> Ve
...
+    for n in 0..10000 {
         assert_eq!(
             FrozenSubTreeIterator::new(n).collect::<Vec<_>>(),
             slow_get_frozen_subtree_roots(n),
         );
     }
 }
 
-fn collect_all_positions(num_leaves: usize, num_new_leaves: usize) -> Vec<u64> {
+fn collect_all_positions(num_leaves: u64, num_new_leaves: u64) -> Vec<u64> {
```
Comment (msmouse) [inline types/src/proof/position/position_test.rs]: why not `LeafCount`?

@bors-libra delegate+

Comment (zamsden) [reply types/src/proof/position/position_test.rs]: I didn't realize it was in scope, but it is so ... done.

**Pull request closed without merging**